external/anomaly/anomaly_classification/task.py

```diff
@@ -0,0 +1,263 @@
+from ote_sdk.entities.model import ModelEntity, ModelPrecision, ModelStatus
+from ote_sdk.entities.resultset import ResultSetEntity
+from ote_sdk.entities.task_environment import TaskEnvironment
+from ote_sdk.entities.train_parameters import TrainParameters
+from ote_sdk.usecases.evaluation.metrics_helper import MetricsHelper
+from ote_sdk.usecases.tasks.interfaces.evaluate_interface import IEvaluationTask
+from ote_sdk.usecases.tasks.interfaces.export_interface import ExportType, IExportTask
+from ote_sdk.usecases.tasks.interfaces.inference_interface import IInferenceTask
+from ote_sdk.usecases.tasks.interfaces.training_interface import ITrainingTask
+from ote_sdk.usecases.tasks.interfaces.unload_interface import IUnload
+from pytorch_lightning import Trainer
+
+logger = logging.getLogger(__name__)
+
+
+class AnomalyClassificationTask(ITrainingTask, IInferenceTask, IEvaluationTask, IExportTask, IUnload):
+    """
+    Base Anomaly Task for Training and Inference
+
+    Args:
+        task_environment (TaskEnvironment): OTE Task environment.
+    """
+
+    def __init__(self, task_environment: TaskEnvironment):
+        self.task_environment = task_environment
+        self.labels = task_environment.get_labels()
+
+        # Hyperparameters.
+        self.project_path: str = tempfile.mkdtemp(prefix="ote-anomalib")
+        self.config = self.get_config()
+
+        self.model_name = task_environment.model_template.name
+        self.model = self.load_model(ote_model=task_environment.model)
+
+        self.trainer: Trainer
+
+    def get_config(self) -> Union[DictConfig, ListConfig]:
+        """
+        Get Anomalib Config from task environment
+
+        Returns:
+            Union[DictConfig, ListConfig]: Anomalib config
+        """
+        hyper_parameters = self.task_environment.get_hyper_parameters()
+        config = get_anomalib_config(ote_config=hyper_parameters)
+        config.dataset.task = "classification"
+        config.project.path = self.project_path
+        return config
+
+    def load_model(self, ote_model: Optional[ModelEntity]) -> AnomalyModule:
+        """
+        Create and Load Anomalib Module from OTE Model.
+        This method checks if the task environment has a saved OTE Model,
+        and creates one. If the OTE model already exists, it returns the
+        the model with the saved weights.
+
+        Args:
+            ote_model (Optional[ModelEntity]): OTE Model from the
+                task environment.
+
+        Returns:
+            AnomalyModule: Anomalib
+                classification or segmentation model with/without weights.
+        """
+        if ote_model is None:
+            model = get_model(config=self.config)
+            logger.info(
+                "No trained model in project yet. Created new model with '%s'",
+                self.model_name,
+            )
+        else:
+            model = get_model(config=self.config)
```

Comment:
Seems that you can move `model = get_model(config=self.config)` outside of the `if-else` clause.